### ISSUE TITLE
VarSet: Value input should show units for properties that have units

### ISSUE THREAD

**Issue by Roy-043** [FreeCAD/FreeCAD#15557]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

If you specify a property type that has units, for example `App::PropertyLength`, in the dialog of the Std_VarSet command the Value input should display units. Specifying 1.0 for the length means 1mm, but that is awkward and confusing if the unit system is US customary or Meter decimal for example.

### Full version info

```shell
OS: Windows 8 build 9600
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.38231 (Git)
Build type: Release
Branch: main
Hash: 29889bfd9a7ba0f0496f2e2699d454ea5fb5a05a
Python 3.11.9, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: C/Default (C) [ OS: Dutch/Netherlands (nl_NL) ]
Installed mods:
```

### Subproject(s) affected?

Core

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by pieterhijma** [comment]
I agree. I try to use the same "editors" as the Property View editors and I assumed the behavior was the same but I notice it isn't. Since it was intended to be the same as in the Property View, this can be considered a bug.
